File: 1207. Unique Number of Occurrences/uniqueOccurrences.cpp
class Solution
{
public:
    bool uniqueOccurrences(vector<int> &arr)
    {

        priority_queue<int> maxh;
        unordered_map<int, int> m;
        for (auto it : arr)
        {
            m[it]++;
        }
        for (auto it : m)
        {
            maxh.push(it.second);
        }
        int prev = -1;
        while (maxh.size() > 0)
        {
            if (prev == maxh.top())
                return false;
            prev = maxh.top();
            maxh.pop();
        }
        return true;
    }
};

File: 1207. Unique Number of Occurrences/uniqueOccurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {

        HashSet<Integer> set = new HashSet<>();
        HashMap<Integer, Integer> map = new HashMap<>();

        for (int num : arr) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }

        for (int count : map.values()) {
            if (set.contains(count)) {
                return false;
            }
            set.add(count);
        }

        return true;

    }
}